cibuildwheel/resources/build-platforms.toml

```diff
@@ -0,0 +1,63 @@
+[tool.cibw.build-platforms]
+linux = [
```

Comment:
Apologies, I might have been a little unclear on this earlier. I think the config file should be-

```toml
[linux]
python_configurations = [
  { identifier = "cp27-manylinux_x86_64", version = "2.7", path_str = "/opt/python/cp27-cp27m" },
  "...",
  "...",
  "...",
]

[macos]
python_configurations = [
  { identifier = "cp27-macosx_x86_64", version = "2.7", url = "https://www.python.org/ftp/python/2.7.18/python-2.7.18-macosx10.9.pkg" },
  "...",
  "...",
  "...",
]

[windows]
python_configurations = [
  { identifier = "cp27-win32", version = "2.7.18", arch = "32" },
  "...",
  "...",
  "...",
]
```

I think it looks neater, and might let us extend with more stuff in the future, if we need.

Reply:
Okay, I like following the pyproject.toml style, but technically, in retrospect, we don't own "cibw" so I don't think we would have been correct in using `tool.cibw`; it needs to be `tool.cibuildwheel`. If we made all configs pyproject.toml compatible, then eventually add the ability for cibuildwheel to pick up pyproject.tom, you could avoid having to pass overrides for these, and instead just use the "one configuration file" as it was meant to be used! :) (Not that I'd expect many/anyone to override this, but pinned docker images specified in pyprjoect.toml could be very useful?)

So is `tool.cibuildwheel.linux`, etc. fine?

Reply:
Why does this need to be here? I can see why you need to namespace in `pyproject.toml`, but this is a purely internal file; I'm not sure if/why it should be namespaced?

Reply:
Oh, OK, you're thinking ahead of trying to read from `pyproject.toml`?

Reply:
Yes, should make the merge easy and be consistent. Though [some tools](https://pycqa.github.io/isort/docs/configuration/config_files/) do keep namepacing out of "custom" files, and only use it in the shared config, so it's not critical. (most tools pre-date toml, as well, so the custom config is .cfg instead). Will wait to update until @joerick weighs in.

Reply:
If we did have `[tool.cibuildwheel]` options in `pyproject.toml`, we'd want those options to be deliberately user-facing - designed, and documented. This seems like it's internal data, not config (and it's very prone to change in schema).

Reply:
Okay, sounds fine to me then. I would expect eventually allowing some things to be set from `[tool.cibuildwheel.*]` to be quite useful, though I don't see a reason to have this ever settable, and certainly not likely in it's current form.

Reply:
@joerick Sorry, one more question. What's a good name for the file? This fine or something else?

Reply:
(Assuming it might depend on what you think could go in here later)

Reply:
This seems good to me :)